enhancements/network/configure-ovs-alternative.md

```diff
@@ -0,0 +1,562 @@
+configuration will be provided through Ignition it will still be necessary
+to have platform-specific methods of injecting network config before Ignition
+runs. However, the platform-specific part can be much simpler because it
+won't need to handle every possible network configuration, just enough to
+reach the Ignition endpoint. In some use cases, platform-specific configuration
+won't even be needed.
+
+### Workflow Description
+
+When the user is populating install-config they will provide the necessary
+configuration in the networking section (TODO: Is this the right place?).
+It will look something like this:
+
+networking:
+  networkType: OVNKubernetes
+  machineNetwork:
+  - cidr: 192.0.2.0/24
+  [...]
+  hostConfig:
+  - name: master-0
+    networkConfig:
+      interfaces:
+      - name: enp2s0
+        type: ethernet
+        state: up
+        ipv4:
+          enabled: false
+        ipv6:
+          enabled: false
+      - name: br-ex
+        type: ovs-bridge
+        state: up
+        copy-mac-from: enp2s0
+        ipv4:
+          enabled: false
+          dhcp: false
+        ipv6:
+          enabled: false
+          dhcp: false
+        bridge:
+          port:
+          - name: enp2s0
+          - name: br-ex
+      - name: br-ex
+        type: ovs-interface
+        state: up
+        ipv4:
+          enabled: true
+          dhcp: true
+        ipv6:
+          enabled: false
+          dhcp: false
+
+The name field will need to be something that uniquely identifies a given host.
+I'm unsure if name is necessarily the best way to do that, but it is one valid
+option. I reused the networkConfig name from the baremetal feature for
+consistency. Perhaps something different would be preferable to avoid confusion
+though?
+
+Note that although the configuration can be provided on a per-host basis, it is
+not mandatory to do so. YAML anchors can be used to duplicate a single
+configuration across multiple nodes. For example:
+
+networking:
+  networkType: OVNKubernetes
+  machineNetwork:
+  - cidr: 192.0.2.0/24
+  [...]
+  hostConfig:
+  - name: master-0
+    networkConfig: &BOND
+      interfaces:
+      - name: br-ex
+        ...etc...
+  - name: master-1
+    networkConfig: *BOND
+  - name: master-2
+    networkConfig: *BOND
+
+This reduces duplication in install-config and the chance of errors due to
+unintended differences in node config.
+
+When this mechanism is in use, a flag will be set to indicate to configure-ovs
+that it should not attempt to configure br-ex.
+
+If changes to the bridge configuration are desired on day 2 (e.g. adding a VLAN
+or changing MTU), the user will need to install Kubernetes-NMState and use it
+to apply those changes directly to br-ex.
+
+#### Variation [optional]
+
+This could be implemented almost entirely with machine-configs, with only a simple
+service on the host to apply the provided NMState files. This is the approach
```

Comment:
The systemd service is already in place as part of nmstate installation, we just need to put yamls at `/etc/nmstate/`

Reply:
It is, but I wonder if that does what we need. I only want this to apply once, and only the configuration that applies to the specific host. If we implemented this with machine-configs, all of the configs for all of the nodes will be present in the directory. I also only want this to be applied at first boot, and then if changes are needed they should be made by kubernetes-nmstate. We _could_ allow updating via machine-config, but then we throw out all of the benefits that kubernetes-nmstate gives us.

Reply:
Stuff under `/etc/nmstate` apply only once, then nmstate move the file so it's not apply at reboot, also I understand that machine config will dump there just the one for that node.

Reply:
Unfortunately there's no way to tell MCO to only deploy the config specific to the node. We have to write all of the configs to all of the nodes and have logic on the node itself to pick the right one. I've added some discussion of how we might integrate with the NMState service to do this in the Variation section.